pandas/plotting/_core.py

```diff
@@ -2951,21 +2951,46 @@ def box(self, by=None, **kwds):
 
     def hist(self, by=None, bins=10, **kwds):
         """
-        Histogram
+        Draw one histogram of the DataFrame's Series using matplotlib.
+
+        A histogram is a representation of the distribution of data.
+        This function groups the values of all given Series in the DataFrame
+        into bins, and draws all bins in only one matplotlib.AxesSubplot. This
+        is useful when the DataFrame's Series are in a similar scale.
 
         Parameters
         ----------
-        by : string or sequence
+        by : str or sequence
             Column in the DataFrame to group by.
-        bins: integer, default 10
-            Number of histogram bins to be used
-        `**kwds` : optional
-            Additional keyword arguments are documented in
-            :meth:`pandas.DataFrame.plot`.
+        bins : int, default 10
+            Number of histogram bins to be used.
+        **kwds : optional
+            Keyword arguments to pass on to :meth:`pandas.DataFrame.plot`.
 
         Returns
         -------
-        axes : :class:`matplotlib.axes.Axes` or numpy.ndarray of them
+        axes : matplotlib.AxesSubplot histogram.
```

Comment:
The "or ndarray of them" was not correct?

Reply:
No, it was not correct. This function always returns one matplotlib.AxesSubplot. It's different than DataFrame.hist(), which returns an ndarray.

Reply:
Just looked, and the key to get this is passing `subplots=True`, then the different columns are each plotted in a subplot in instead of in overlay, en then you get an ndarray of axes

Reply:
I see. However this is actually a parameter that is passed on to the plot() function, but it's not part of the default plot.hist() parameters... so IMHO it's ok to say that the return will be just one. Don't you think so?

Reply:
Yes, we should still document it here, since this is what the return value of this function can be. 

(btw, you are welcome to add `subplots` to the list of Parameters, we have a general issue about that we should document more of those common parameters in each of the methods)